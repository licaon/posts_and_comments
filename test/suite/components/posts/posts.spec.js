import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Posts from 'src/components/posts/posts';

describe('Posts component', () => {
  const createComponent = (props) => {
    const wrapper = shallow(
      <Posts
        {...props}
      />);
    return toJson(wrapper);
  };
  it('should render component correctly without posts', (done) => {
    const component = createComponent();

    expect(component).toMatchSnapshot();
    done();
  });
  it('should render component correctly with posts', (done) => {
    const component = createComponent({
      posts: [
        {
          id: 1,
        },
      ],
    });

    expect(component).toMatchSnapshot();
    done();
  });
});
