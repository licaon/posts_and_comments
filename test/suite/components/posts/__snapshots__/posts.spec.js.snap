// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts component should render component correctly with posts 1`] = `
<div
  className="md-cell--8"
>
  <Connect(Post)
    post={
      Object {
        "id": 1,
      }
    }
  />
</div>
`;

exports[`Posts component should render component correctly without posts 1`] = `
<div
  className="md-cell--8"
/>
`;
